grails-app/domain/org/pih/warehouse/inventory/CycleCount.groovy

```diff
@@ -77,29 +77,25 @@ class CycleCount {
      * Determines what the CycleCountStatus should be for the cycle count.
      */
     CycleCountStatus recomputeStatus() {
-        if (!cycleCountItems || cycleCountItems.every { it.status == CycleCountItemStatus.READY_TO_COUNT }) {
-            return CycleCountStatus.REQUESTED
-        }
-        if (cycleCountItems.every { it.status == CycleCountItemStatus.REVIEWED }) {
-            return CycleCountStatus.REVIEWED
-        }
-        if (cycleCountItems.every { it.status == CycleCountItemStatus.READY_TO_REVIEW }) {
-            return CycleCountStatus.READY_TO_REVIEW
+        Set<CycleCountItem> items = itemsOfMostRecentCount
+        if (items.every { it.status.isCompleted() }) {
+            return CycleCountStatus.COMPLETED
         }
-        if (cycleCountItems.any { it.status == CycleCountItemStatus.INVESTIGATING }) {
+        if (items.any { it.status == CycleCountItemStatus.INVESTIGATING }) {
             return CycleCountStatus.INVESTIGATING
         }
-        if (cycleCountItems.any { it.status == CycleCountItemStatus.COUNTING }
-                && !cycleCountItems.any { it.status in [CycleCountItemStatus.INVESTIGATING, CycleCountItemStatus.READY_TO_REVIEW] }) {
+        if (items.any { it.status == CycleCountItemStatus.COUNTING }) {
```

Comment:
@kchelstowski heads up that I put the COUNTING condition before the COUNTED one. I wasn't going to push any changes here but I needed to fix this to test other things locally so I figured I'd push straight to your branch and (hopefully) save you some headache.

This change made sense to me because if a count has one item that was counted but one that is not finished being counted yet, the count should still be in the COUNTING state because we're not done counting.

Let me know if you disagree though.

Reply:
That's fine @EWaterman. Thanks for the effort